eng/pipelines/templates/steps/generate-doc.yml

```
@@ -15,7 +15,17 @@ steps:
   - pwsh: |
       node $(Build.SourcesDirectory)/eng/tools/generate-doc/dist/index.js --serviceDir "${{parameters.ServiceDirectory}}"
     displayName: "Run Typedoc Docs"
+    condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
+
+  - pwsh: |
+      if ("$(ChangedServices)") {
```

```suggestion
      if ('$(ChangedServices)' -and '$(ChangedServices)' -notlike '*ChangedServices*') {
```

These types of statements always evaluate to true or cause a powershell syntax error because if devops doesn't set the variable it just gets rendered as the literal '$(var)'. So single quotes prevents the syntax error, and a second evaluation determines whether it's unset.